tests/test/object_safe.rs

```diff
@@ -0,0 +1,29 @@
+use super::*;
+
+/// The current implementation of RustIrDatabase::is_object_safe only checks for
+/// the #[object_safe] flag. This test will start failing once it performs a
+/// proper test because Foo should be object safe.
```

Comment:
Is this explanation correct? That is, will chalk eventually check trait definitions to determine object safety, or will rustc just give us that information?

Reply:
If we *do* get to a point where Chalk is checking for object safety itself, it won't be for a bit (probably). But, I'm not sure how helpful this comment is. But that's also somewhat in part because I'm not sure what tests we *do* want in this file.